src/gevent/monkey.py

```diff
@@ -255,11 +256,18 @@ def get_original(mod_name, item_name):
 _NONE = object()
 
 
-def patch_item(module, attr, newitem):
+def patch_item(module, attr, newitem, _patch_module=False):
     olditem = getattr(module, attr, _NONE)
     if olditem is not _NONE:
         saved.setdefault(module.__name__, {}).setdefault(attr, olditem)
     setattr(module, attr, newitem)
+    if _patch_module:
```

Comment:
I'm not sure this is a good idea in general. One example that springs to mind is pickling; another is that it may tend to produce very confusing `repr` results. Why would we want to do this?

If this is something that's for very specialized use-cases, perhaps that would be better served in the hook functions for the module? (See below for some general comments about underscore-prefixed things.)

Reply:
Reviewing #1447 I see that pickling was specifically mentioned as the motivation for this. I'm still not convinced, I'd need to see a specific example to understand what problem this solves that can't be solved by the object itself.